beeai/agents/rebase_agent.py

```diff
@@ -177,9 +244,26 @@ async def run_rebase_agent(state):
             finally:
                 os.chdir(cwd)
             if state.rebase_result.success:
+                return "run_build_agent"
+            return "comment_in_jira"
+
+        async def run_build_agent(state):
+            response = await build_agent.run(
+                prompt=build_render_prompt(
+                    BuildInputSchema(
+                        srpm_path=state.rebase_result.srpm_path,
+                        dist_git_branch=state.dist_git_branch,
+                        jira_issue=state.jira_issue,
+                    ),
+                ),
+                expected_output=BuildOutputSchema,
+                execution=get_agent_execution_config(),
+            )
+            build_result = BuildOutputSchema.model_validate_json(response.answer.text)
+            if build_result.success:
                 return "commit_push_and_open_mr"
-            else:
-                return "comment_in_jira"
+            state.build_error = build_result.error
+            return "run_rebase_agent"
```

Comment:
neat 🚀 

how do we prevent here the agent to infinite loop?

Reply:
That's a good question :sweat_smile: I suppose we have to add some counter and give up after a number of failed attempts.

Reply:
sounds good